### DISCUSSION TITLE
Support GUI connect/manage to docker/cli servers?

### DISCUSSION THREAD

**Discussion by ace90210** [mockoon/mockoon#738] [Ideas]
So i recently discovered Mockoon and it turns out it has 99% of the features i want and that missing 1% feels like it probably exists i just cant find it. Unfortunatelly  that last bit is a major requirement for my planned use case.

What im after is two features:

1. (nice to have) Ability to automatically create the mock response while in proxy mode during runtime without user intervention and just put it in a turned off state until the user disables proxy mode (globally or somehow for just that request)
2. Ability to run mockoon servers remotely in a docker container (seems supported) but i also want to manage one or more instances of this via the GUI (which seems to only work on its own servers running locally)

The use case:

The use case i have is many microservices with many dependancies between them. What i want (and have built a working prototype of myself its just not as developed as mockoon) is a mock service with build in proxy that automatically records all traffic while in proxy mode so that it in effect learns how the services and apis respond. i then want to remotely control this mock service at runtime to instantaniously edit the mock data or change if its using the real or mock service.

This allows me to train a vast mock database on scale so that when working on microserver A, B and C of a list of 30 services i can just turn the other services into "mock mode" until i actually need to update the responses from them. 

Initially during uptake i can run it in "transparent mode" aka proxy mode where every service gets the real data and responses until all the typical common responses are populated in the mock.

Mockoon seems to do everything i just cant manage the servers once i configure them unless i run it locally using the desktop UI but thats too limiting (plus i have to manually go to the log, click response and click add on every request i want it to "learn" which on the scale i see me using this its not feasible)

Also i work in a enterprise company who would possible pay well for this service if it supported the above and i can demo it working on their system (which uses microk8 but i think that shouldnt be an issue as mockoon supports docker)

any suggestions or questions please feel free to ask and a demo can be made of my prototype system which supports all this its just missing many features of mockoon like the fakejs (though it has many features for scalability mockoon lacks as well such as multi tenancy, grouping of services(servers in mockoon) etc)

**Comment by 255kb** [comment]
Concerning your no. 1, there is currently no "auto recording" mode that lets you call proxied endpoints and create the corresponding mocks. This was requested from time to time in the past but it is not currently planned. Feel free to open an issue to track the users interest for this feature.

Concerning the 2, it's something that I have in my mind since a while and probably something I would like to explore in a near future. Managing a local or remote CLI with the desktop application seems to be a normal evolution. I will probably create an RFC for this kind of feature in the coming days.

**Comment by yippie** [comment]
I would like to add my vote for managing a remote CLI with the desktop application. It would make standing up and using a long term cli instance much more manageable.
